app/grandchallenge/components/models.py

```diff
@@ -2219,18 +2223,25 @@ def remove_civ(self, *, civ):
 
     def create_civ(self, *, civ_data, user=None, linked_task=None):
         if not self.is_editable:
-            raise RuntimeError(
-                f"{self} is not editable. CIVs cannot be added or removed from it."
+            # This can happen for Jobs with multiple inputs,
+            # if another input has already failed validation
+            logger.error(
+                f"{self} is not editable. CIVs cannot be added or removed from it.",
+                exc_info=True,
             )
+            return
```

Comment:
For my understanding, what was the motivation for this change? We're definitely okay to continue here?

Reply:
This was a bug in the implementation that we just have not hit yet in production. You can run into this If a job has multiple inputs, say 2 image inputs, and the user selects existing images for both where the first fails validation (voxel values don't match, for example). The job is then marked as CANCELLED before the second input is validated. 

This is a corner case that hardly ever happens because people are probably more likely to upload new images then to search for an existing image. For values, where validation is also instantaneous (not async), this is not an issue because they are validated client side already.

Reply:
Another way of circumventing this would be to subset the choice of existing images by interface (like we do for files now), but the lookup is a bit more complex for images and I'm not sure if we want to restrict it in that way.

Reply:
I don't think we want to add that restriction. I'm just not sure if not raising the error here is and just returning the right thing we want.

Reply:
We're still reporting the error to sentry here, so we will still get informed (also good because if this happens for display sets or archive items, we need to investigate). If we raise the error, the user will get an ugly error screen. They should just see that the Job has been cancelled. 

What would you do instead?

Reply:
I'm not sure which is why I'm asking. Should it raise an exception and be handled in the caller? I would expect a function called `create_civ` to do exactly that, and something must have gone wrong if it didn't. I think the risk is here that the caller function carries on having expected a civ to be created but it wasn't.